File: sandbox.controller/src/kz/greetgo/sandbox/controller/model/ClientRecord.java
package kz.greetgo.sandbox.controller.model;

import java.time.LocalDate;
import java.time.Period;
import java.time.ZoneId;

public class ClientRecord {
  public String fio;
  public Character character;
  public int age;
  public float totalBalance;
  public float minBalance;
  public float maxBalance;
  public int clientId;

  public ClientRecord() {
  }

  public ClientRecord convertToSaveToClientRecord(ClientToSave toSave) {
    this.fio = toSave.surname + " " + toSave.name;
    if (toSave.patronymic != null)
      this.fio += " " + toSave.patronymic;
    LocalDate currentDate = LocalDate.now();
    LocalDate birthDate = toSave.birthDay.toInstant()
      .atZone(ZoneId.systemDefault())
      .toLocalDate();
    if ((birthDate != null) && (currentDate != null)) {
      this.age = Period.between(birthDate, currentDate).getYears();
    } else {
      this.age = 0;
    }

    this.character = toSave.character;
    this.clientId = toSave.clientID;
    return this;

  }
}

File: sandbox.client/src/model/ClientRecord.ts
import {Character} from "./Character";

export class ClientRecord {
  public fio: string;
  public character: Character;
  public age: number/*int*/;
  public totalBalance: number/*int*/;
  public minBalance: number/*int*/;
  public maxBalance: number/*int*/;
  public clientId: number/*int*/;

  public static create(a: any): ClientRecord {
    const ret = new ClientRecord();
    ret.assign(a);
    return ret;
  }

  assign(a: any) {
    this.age = a.age;
    this.fio = a.fio;
    this.character = a.character;
    this.maxBalance = a.maxBalance;
    this.totalBalance = a.totalBalance;
    this.minBalance = a.minBalance;
    this.clientId = a.clientId;
  }
}